Fixed issues with levels and switch

slavvka opened the pull request:
Correctly fixes #48

slavvka commented:
@Hanh94 Could you please review?

Hanh94 commented:
Hi @slavvka ,Thanks for your PR.
We should have solved the adaptation problem of fans in the latest version, and adopted some of your optimization suggestions. Thanks again.

slavvka commented:
@Hanh94 Should I close this PR?

slavvka commented:
@Hanh94 I resolved the conflicts in case you still need the fix

slavvka commented:
@Hanh94 the current version can only turn on and of my fan but cannot control the rotation speed. In this fix I modified the code to take the real list of possible values for the speed and use them. Also I added support for `fan_speed_enum` control since my fan does use it instead of `fan_speed` despite it exists in its config.

slavvka commented:
@Hanh94 I resolved the conflicts again. Could you please review it? I really want to have a possibility to control my fan's rotation speed.

slavvka commented:
@Hanh94 I also fixed the light for Fanv2 accessory.

mattwelch commented:
I was unable to control my Tuya-connected [Treatlife Fan](https://www.amazon.com/dp/B08P5D3ZKW?psc=1&ref=ppx_yo2_dt_b_product_details) switch's speed. I merged the above PR to my local repo, and things are working great now. Would love to see this in the official release.

slavvka commented:
@Hanh94 Is there any news?

slavvka commented:
@zlinoliver Could you please review and merge this PR?

slavvka commented:
@Hanh94 Any news?

slavvka commented:
@Hanh94 @zlinoliver Guys, why are you ignoring this PR?

slavvka commented:
@Hanh94 @zlinoliver Guys, why are you ignoring this PR?

dfizzo commented:
@slavvka Did you ever get this resolved? I have the same issue and would love to get a fix in place. Thanks!

slavvka commented:
> @slavvka Did you ever get this resolved? I have the same issue and would love to get a fix in place. Thanks!

Unfortunately the owners keep ignoring my PR :(

mattwelch commented:
Just noting here that this PR continues to fix the speed control problem w/ fanv2. I had lost that control when I updated, and it took me a few days to remember that I had to manually apply this PR. Would love to see it officially merged!

bFollon commented:
Having this stagnant PR here really discourages creating new ones...

dfizzo commented:
So frustrating indeed
________________________________
From: bFollon ***@***.***>
Sent: Friday, February 24, 2023 5:58:08 PM
To: tuya/tuya-homebridge ***@***.***>
Cc: dfizzo ***@***.***>; Comment ***@***.***>
Subject: Re: [tuya/tuya-homebridge] Correct support of speed levels and light for Fanv2 (#77)


Having this stagnant PR here really discourages creating new ones...

—
Reply to this email directly, view it on GitHub<https://github.com/tuya/tuya-homebridge/pull/77#issuecomment-1444621987>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AT52XE7E6CXYT5QH75ZX7SDWZE4IBANCNFSM5BG4VOFA>.
You are receiving this because you commented.Message ID: ***@***.***>

0x5e commented:
they are already fixed on my fork, if anything not working for fan, you might using the non-standard device, need config manually, see my readme.

dfizzo commented:
Thanks @gaosen. Will have a look at the fork!_________________________________________Daniel M. ***@***.***_________________________________________On Feb 24, 2023, at 7:14 PM, gaosen ***@***.***> wrote:﻿
they are already fixed on my fork, if anything not working for fan, you might using the non-standard device, need config manually, see my readme.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>
